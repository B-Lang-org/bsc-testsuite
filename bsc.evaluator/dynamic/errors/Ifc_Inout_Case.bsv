interface Ifc;
   interface Inout#(Bool) io_out;
endinterface

(* synthesize *)
module sysIfc_Inout_Case
          (Inout#(Bool) io1, Inout#(Bool) io2,
           Inout#(Bool) io3, Inout#(Bool) io4,
           Ifc ifc);

   Reg#(Bit#(2)) idx <- mkReg(0);

   Inout#(Bool) io;
   case (idx)
      0 : io = io1;
      1 : io = io2;
      2 : io = io3;
      default : io = io4;
   endcase

   interface io_out = io;

endmodule

