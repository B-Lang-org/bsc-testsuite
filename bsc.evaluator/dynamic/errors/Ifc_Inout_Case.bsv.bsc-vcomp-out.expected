checking package dependencies
compiling Ifc_Inout_Case.bsv
code generation for sysIfc_Inout_Case starts
Error: "Ifc_Inout_Case.bsv", line 14, column 4: (G0094)
  The expression for a inout could not be evaluated. Inouts must be determined
  at compile time and cannot be computed dynamically.
  During elaboration of the interface inout `io_out' at "Ifc_Inout_Case.bsv",
  line 6, column 8.
  During elaboration of `sysIfc_Inout_Case' at "Ifc_Inout_Case.bsv", line 6,
  column 8.
